### PR TITLE
[merged] pull: Free fetch_data by default

### PR DESCRIPTION
**Pull request by Mathnerd314**
This should fix the memory leaks in #352

This is a subset of the changes, the other part is in my pull code rewrite

### CODE REVIEW

File: src/libostree/ostree-repo-pull.c [hunk 1]
```diff
@@ -761,11 +762,17 @@ content_fetch_on_complete (GObject        *object,
                                        object_input, length,
                                        cancellable,
                                        content_fetch_on_write_complete, fetch_data);
+      free_fetch_data = FALSE;
     }
 
  out:
   pull_data->n_outstanding_content_fetches--;
   check_outstanding_requests_handle_error (pull_data, local_error);
+  if(free_fetch_data)
```
Comment (cgwalters) [inline src/libostree/ostree-repo-pull.c]: Space between `if` and paren `(` please.

Comment (Mathnerd314) [inline src/libostree/ostree-repo-pull.c]: Done.

**Comment by cgwalters** [comment]
@rh-atomic-bot r+ 2e40aa2

**Comment by cgwalters** [comment]
Thanks for splitting this out cleanly!

**Pull request closed without merging**